client/app/cloud/project/compute/infrastructure/virtualMachine/add/cloud-project-compute-infrastructure-virtualMachine-add.service.js

```diff
@@ -123,25 +127,35 @@ class CloudProjectVirtualMachineAddService {
   }
 
   groupFlavorsByCategory(flavors, flavorsTypes) {
-    const categorizedFlavors = [];
-    _.forEach(flavorsTypes, (flavorType) => {
-      const category = this.CloudFlavorService.getCategory(flavorType, true);
-      const filteredFlavor = _.filter(flavors, { type: flavorType });
-      if (filteredFlavor.length > 0) {
-        const categoryObject = _.find(categorizedFlavors, { category: category.id });
-        if (categoryObject) {
-          categoryObject.flavors = _(categoryObject.flavors)
-            .concat(_.filter(flavors, { type: flavorType })).value();
-        } else {
-          categorizedFlavors.push({
+    return _(flavorsTypes)
+      .chain()
+      .reduce((previousValues, flavorType) => {
+        const flavorsOfCurrentFlavorType = _(flavors).filter({ type: flavorType }).value();
```

Comment:
Implicit chaining isn't helpful in that case.

```js
const flavorsOfCurrentFlavorType = _.filter(flavors, { type: flavorType });
```

Reply:
I like this form better